### DISCUSSION TITLE
How to pin Java version in `bin.write_jar_script`?

### DISCUSSION THREAD

**Discussion by devton-here** [Homebrew/discussions#3591] [Writing Formulae/Casks] [answered]
I have this form: https://github.com/heremaps/homebrew-olp-cli/blob/main/Formula/olp.rb

Basiscally:

```
    libexec.install "cli_2.12-12.2.5.jar"
    bin.write_jar_script libexec/"cli_2.12-12.2.5.jar", "olp", "-Dfile.encoding=UTF8"
```

and as far as I understand the second line is creating the script at `/usr/local/bin/olp` with this content:

```
#!/bin/bash
export JAVA_HOME="${JAVA_HOME:-/usr/local/opt/openjdk@8/libexec/openjdk.jdk/Contents/Home}"
exec "${JAVA_HOME}/bin/java" -Dfile.encoding=UTF8 -jar "/usr/local/Cellar/olp@10.5/10.5.125/libexec/cli_2.12-10.5.125.jar" "$@" 
```

Now I have this issue: some people upgraded their Java version and also JAVA_HOME was updated which lead to errors, because the jar is not yet compatible with some newer Java versions.

I would like to generate a script with this content or something like it:
```
#!/bin/bash
export JAVA_HOME="/usr/local/opt/openjdk@8/libexec/openjdk.jdk/Contents/Home"
exec "${JAVA_HOME}/bin/java" -Dfile.encoding=UTF8 -jar "/usr/local/Cellar/olp@10.5/10.5.125/libexec/cli_2.12-10.5.125.jar" "$@" 
```

How can I do this?

**Comment by SMillerDev** [comment]
Like https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/embulk.rb#L25

**Reply by devton-here** [reply]
I've tried that exactly, see https://github.com/heremaps/homebrew-olp-cli/pull/2/files

but it get even worse, because there is no default for JAVA_HOME anymore:

```
#!/bin/bash
export JAVA_HOME="${JAVA_HOME:-}"
exec "${JAVA_HOME}/bin/java" -Dfile.encoding=UTF8 -jar "/usr/local/Cellar/olp/12.2.5/libexec/cli_2.12-12.2.5.jar" "$@"
```

**Reply by carlocab** [reply]
Have you tried using `java_version: "1.8"`?

**Reply by devton-here** [reply]
> Have you tried using java_version: "1.8"?

You mean to set it to `"1.8"` for `openjdk@8`?

**Reply by carlocab** [reply]
Yes. See: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/embulk.rb#L25

**Comment by Bo98** [comment, marked as answer]
`write_jar_script` doesn't support non-overridable JAVA_HOME envs currently.

The closest you'll get is to implement it manually, e.g. like https://github.com/Homebrew/homebrew-core/blob/0a4254c2582babf2538d3bf9760a8eca05dff196/Formula/yeti.rb?q=%22-jar%22#L27 but with `java_home_env` instead of `overridable_java_home_env`
